### ISSUE TITLE
Excluding Configuration Files from Merge when Merging Remote Branches on Gitea

### ISSUE THREAD

**Issue by SeesawLiu** [go-gitea/gitea#26682]
### Description

I'm currently using Gitea and I'm facing a situation where I need to merge two remote branches – one is the development branch, and the other is the release branch. While merging from the development branch to the release branch, I want to exclude certain configuration files from being merged. The release branch is currently configured for only online merges.

To achieve this, I read through the following links:

[Git Documentation - Merge Strategies](https://git-scm.com/docs/merge-strategies)
[Git Book - Customizing Git Attributes - Merge Strategies](https://git-scm.com/book/en/v2/Customizing-Git-Git-Attributes#_merge_strategies)

I intended to use the .gitattributes file to control the merge strategy, but it seems that the configuration hasn't taken effect as expected. I'm looking for guidance on how to successfully implement the requirement mentioned above.

### Details:

Development Branch: <development-branch>
Release Branch: <release-branch>
Configuration Files: <config-file-1>, <config-file-2>

### Steps I've Taken:

Created a .gitattributes file in the root directory of the repository.
Added the following lines to the .gitattributes file:
``` arduino
    <config-file-1> merge=ours
    <config-file-2> merge=ours
```
### Expected Outcome:

When I perform a merge from <development-branch> to <release-branch>, I expect the specified configuration files <config-file-1> and <config-file-2> to be excluded from the merge and to keep the versions from the release branch.

### Actual Outcome:

Despite adding the configurations to the .gitattributes file, the merge of the configuration files still occurs when merging branches.

### Request:

I'm seeking assistance in resolving this issue and ensuring that the .gitattributes file is correctly configured to exclude certain configuration files from being merged during the branch merge process on Gitea. Your help would be greatly appreciated!

### Gitea Version

1.20.3

### Can you reproduce the bug on the Gitea demo site?

Yes

### How are you running Gitea?

run local

### Database

None

**Comment by lunny** [comment]
You can exclude the files on protected branch of release branch. But that will only return error if you changed the special files in development branch but not ignore the changed special files and merge others.

**Comment by SeesawLiu** [comment]
@lunny yes,I found the function,but it's not I want.I've discovered that merge strategies can achieve the desired outcome, but it appears that Gitea might not support it.

**Comment by lunny** [comment]
I think maybe you can try to set that manually. Gitea has himself git config file in `data/home` directory.

**Comment by SeesawLiu** [comment]
Adding configuration files in /data/home is effective. thank you.
I'm unsure if this will have an impact on Gitea.

**Comment by dast5022** [comment]
HI @SeesawLiu, did you find a solution for your problem, you could share with us. Or didn´t I understand this diskussion and the solution is already here. Can you help me? I also like to exclude files from merges.

**Comment by wxiaoguang** [comment]
`[git.config]` in `app.ini`
